Update links and replace/remove deprecated modules

Totktonada opened the pull request:
The list of changes:

* Updated broken tarantool.org links.
* Updated HTTP client and server info.
* Replaced 'prometheus' with 'metrics'.
* Removed ljsyscall.
* Removed Swift connector.

See commit messages for details.

Supersedes PR #6.

Totktonada committed:
Update broken tarantool.org links

Supersedes PR #6
See https://github.com/tarantool/doc/issues/2336

Totktonada committed:
Update HTTP client and server info

The story behind this change is the following.

Initially the 'http' module provides a client and a server both. In 2016
the alternative libcurl based client was implemented as the 'curl'
module.

In 2017 we [re-implemented][1] the libcurl based HTTP client as built-in
tarantool module (it was mainly to implement feedback daemon on top of
it).

Later in this year we [removed][2] the HTTP client from the 'http'
module. Now it contains only the server.

In 2019 we forked the curl repository into tarantool and moved the old
'curl' module into the 'tarantool-curl' repository (it was part of
[this][3] task).

Now the HTTP client is built-in and may be called with
`require('http.client')`. The server is in the external 'http' module
and available as `require('http.server')` just as before. This state is
described in the awesome-tarantool list now.

[1]: https://github.com/tarantool/tarantool/issues/2083
[2]: https://github.com/tarantool/http/issues/47
[3]: https://github.com/tarantool/tarantool/issues/4318

Totktonada committed:
Replace 'prometheus' with 'metrics'

The prometheus module is deprecated and now we recommend to use the
metrics module.

Totktonada committed:
Remove ljsyscall

The link is dead: the tarantool-contrib GitHub organization has only
ljsyscall fork for a long time, but now it is gone. The upstream is
here: https://github.com/justincormack/ljsyscall

I don't list the upstream project instead of the fork, because I never
heard about usages on the module within tarantool's application or
module. I don't ever see any relevant questions or discussions.

Despite the module looks useful for implementing low level things in
Lua, it does not look relevant for this list ATM.

pgulutzan commented:
@Totktonada:

Spelling error: Graphite/Prometeus should be Graphite/Prometheus.

Your comment on the "Update broken tarantool.org links" edit says:
"There is no direct match for the old 'box' link ..."
I think there is a direct match. The old link was:
https://tarantool.org/doc/book/box/index.html
the closest thing to that is
https://www.tarantool.io/en/doc/latest/book/box/
rather than
https://www.tarantool.io/en/doc/latest/book/box/data_model/

You have replaced all dead links, except this one for Swift connector:
https://github.com/tris-foundation/tarantool
Should it be removed?

You have removed the entry for curl - libcurl bindings for Tarantool.
- [curl](https://github.com/tarantool/curl) -
  libcurl bindings for Tarantool.

Was that deliberate?

I assume you are aware that the first link
to DataGrip https://www.jetbrains.com/datagrip/
points to a page with a list of 21 supported database management systems
-- and Tarantool is not on the list, it's advertising the competition.
But I agree that is okay, as I don't see a better choice.

Totktonada committed:
Remove Swift connector

It seems, it is gone.

See https://github.com/tarantool/doc/issues/2337

Totktonada commented:
> Spelling error: Graphite/Prometeus should be Graphite/Prometheus.

Thanks, fixed.

----

> I think there is a direct match. The old link was:
> https://tarantool.org/doc/book/box/index.html
> the closest thing to that is
> https://www.tarantool.io/en/doc/latest/book/box/

Looks so. There are several problems around this book:

- Is there a way to navigate inside except the direct link?
- The navigation over the book looks broken (Data model --(next)--> box.space API).
- It is unclear, whether the documentation team intend to support it and fix problems.

Anyway, it is available ATM, so I'll use this link instead of 'Data model'. I'll keep my eyes on it.

----

> You have replaced all dead links, except this one for Swift connector:
> https://github.com/tris-foundation/tarantool
> Should it be removed?

Seems so. I filed the similar [issue](https://github.com/tarantool/doc/issues/2337) against the documentation team. Removed in this PR.

----

> You have removed the entry for curl - libcurl bindings for Tarantool.
> - [curl](https://github.com/tarantool/curl) -
>   libcurl bindings for Tarantool.
>
> Was that deliberate?

I should describe it better in the commit message. The previous message:

> **Update HTTP client and server info**
>
> The client now bundled into tarantool. The tarantool/http module now
> provides only HTTP server.

Now:

> **Update HTTP client and server info**
>
> The story behind this change is the following.
>
> Initially the 'http' module provides a client and a server both. In 2016
> the alternative libcurl based client was implemented as the 'curl'
> module.
>
> In 2017 we [re-implemented][1] the libcurl based HTTP client as built-in
> tarantool module (it was mainly to implement feedback daemon on top of
> it).
>
> Later in this year we [removed][2] the HTTP client from the 'http'
> module. Now it contains only the server.
>
> In 2019 we forked the curl repository into tarantool and moved the old 
> 'curl' module into the 'tarantool-curl' repository (it was part of
> [this][3] task).
>
> Now the HTTP client is built-in and may be called with
> `require('http.client')`. The server is in the external 'http' module
> and available as `require('http.server')` just as before. This state is
> described in the awesome-tarantool list now.
>
> [1]: https://github.com/tarantool/tarantool/issues/2083
> [2]: https://github.com/tarantool/http/issues/47
> [3]: https://github.com/tarantool/tarantool/issues/4318

I hope it explains the reason of the change.

----

> I assume you are aware that the first link
> to DataGrip https://www.jetbrains.com/datagrip/
> points to a page with a list of 21 supported database management systems
> -- and Tarantool is not on the list, it's advertising the competition.
> But I agree that is okay, as I don't see a better choice.

You're right. But I don't see a better way too. I notified @Teinnsei and @Periecle just in the case.

pgulutzan reviewed (approved):
Looks good to me now. Thank you.

Totktonada commented:
Thanks for the careful review!
